src/klein/_message.py

```diff
@@ -0,0 +1,160 @@
+# -*- test-case-name: klein.test.test_message -*-
+# Copyright (c) 2017-2018. See LICENSE for details.
+
+"""
+HTTP message API.
+"""
+
+from typing import Any, Optional, Union
+
+from attr import attrib, attrs
+from attr.validators import instance_of, optional
+
+from tubes.itube import IFount
+
+from twisted.internet.defer import Deferred, succeed
+
+from zope.interface import Attribute, Interface
+
+from ._headers import IHTTPHeaders
+from ._tubes import bytesToFount, fountToBytes
+
+Any, Deferred, IFount, IHTTPHeaders, Optional  # Silence linter
+
+
+__all__ = ()
+
+
+
+# Interfaces
+
+class NoFountError(Exception):
+    """
+    Request has no fount.
+    This implies someone already accessed the body fount.
```

Comment:
I think NoFound is not descriptive enough. Why not "FoundAlreadyAccessed" error?

Reply:
OK, 3ce80d8cc251276f0cb66d83b83a6472b5f1d84e